FirebaseTestProject/Views/PostRow.swift

```diff
@@ -8,25 +8,92 @@
 import SwiftUI
 
 struct PostRow: View {
-    let post: Post
+    @Binding  var post: Post
+    @State var isPresenting: Bool = false
+    @State private var showAlert: Bool = false
+    @State private var systemImage: String = ""
+    let deletePostAction:((Post) -> Void)
+    let isFavoriteAction:((Post) -> Void)
     
+    init(post: Binding<Post>, deletePostAction: @escaping (Post) -> Void, isFavoriteAction: @escaping (Post) -> Void) {
+        self._post = post
+        self.deletePostAction = deletePostAction
+        self.isFavoriteAction = isFavoriteAction
+    }
     
-   var body: some View {
-         VStack {
-             Text(post.title)
-                 .font(.title)
-                 .padding(.bottom, 8)
-             Text(post.text)
-                 .font(.body)
-                 .padding(.bottom, 8)
-             Text(post.author)
-                 .padding(.bottom, 8)
-         }
-   }
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                Text(post.title)
+                    .font(.largeTitle)
+                    .padding(.leading)
+                Spacer()
+                let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
```

Comment:
Please see the note above regarding handling the user ID this way. I was thinking providing it as an environment key might work a little better – just an idea.